umbrella-feeds/src/lib.rs

```diff
@@ -0,0 +1,730 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+//! # Umbrella feeds
+//!
+//! Main contract for all on-chain data.
+//! Check `UmbrellaFeedsReader` to see how to integrate.
+//!
+//! ATTENTION: Keep the `upgradeNatively`/`unregister` entry points in this contract at all time and make sure their logic can be
+//! executed successfully via an invoke to the `atomicUpdate` entry point in the`registry` contract. Otherwise you will not be able to
+//! natively upgrade this contract via the `registry` contract anymore.
+//! Using the native upgradability mechanism for this contract is necessary to not break the `UmbrellaFeedsReader` contracts
+//! which include references to this `UmbrellaFeeds` contract.
+use concordium_std::*;
+use core::fmt::Debug;
+
+/// The name of this contract.
+const NAME: &str = "UmbrellaFeeds";
+
+/// The concept of a CHAIN_ID does not exist on Concordium but kept for consistency.
+const CHAIN_ID: u16 = 0;
```

Comment:
let's use `0xC04C`, so it is not 0.

Reply:
Done and changed it using the equivallent decimal number:

const CHAIN_ID: u16 = 49228;

@DZariusz I just wonder if there is a special meaning behind this magic number.

Reply:
`chain_id` has been removed.